src/OrchardCore/OrchardCore.DisplayManagement/Implementation/DefaultHtmlDisplay.cs

```diff
@@ -50,7 +50,19 @@ public async Task<IHtmlContent> ExecuteAsync(DisplayContext context)
             // Check if the shape is pre-rendered.
             if (shape is IHtmlContent htmlContent)
             {
-                return htmlContent;
+                if (htmlContent is PositionWrapper wrapper)
+                {
+                    while (wrapper.Value is PositionWrapper)
+                    {
+                        wrapper = (PositionWrapper)wrapper.Value;
+                    }
+
+                    return wrapper.Value;
+                }
+                else
```

Comment:
you can remove the else block and just return instead

Reply:
Removed